acvm-repo/acvm/src/compiler/transformers/mod.rs

```diff
@@ -91,7 +91,16 @@ pub(super) fn transform_internal<F: AcirField>(
 ///
 /// Accepts an injected `acir_opcode_positions` to allow transformations to be applied directly after optimizations.
 ///
-/// Does a single optimization pass.
+/// If the width is unbounded, it does nothing.
+/// If it is bounded, it first performs the 'CSAT transformation' in one pass, by creating intermediate variables when necessary.
+/// This results in arithmetic opcodes having the required width.
+/// Then the 'eliminate intermediate variables' pass will remove any intermediate variables (for instance created by the previous transformation)
+/// that are used in exactly two arithmetic opcodes.
+/// This results in arithmetic opcodes having linear combinations of potentially large width.
+/// Finally, the 'range optimization' pass will remove any redundant range opcodes.
+/// It is expected that the backend will be able to handle linear combinations of 'unbounded width' in a more efficient way
+/// than the 'CSAT transformation'.
+/// If it is not the case, the 'eliminate intermediate variables' pass should not be applied.
```

Comment:
I am a bit confused by this comment. Don't we always run the eliminate intermediate variables pass for CSAT or are referring to when we have an unbounded width?

Reply:
Until we have a setting to disable this pass, for now we could just add a brief comment that this pass currently assumes the backend can handle the output of this transformation and thus it is performed.